pkg/controller/syncset/syncset_controller_test.go

```diff
@@ -0,0 +1,175 @@
+package syncset
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	cm "github.com/open-policy-agent/gatekeeper/v3/pkg/cachemanager"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/controller/config"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/controller/config/process"
+	syncc "github.com/open-policy-agent/gatekeeper/v3/pkg/controller/sync"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/fakes"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/readiness"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/syncutil"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/watch"
+	testclient "github.com/open-policy-agent/gatekeeper/v3/test/clients"
+	"github.com/open-policy-agent/gatekeeper/v3/test/testutils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var (
+	configMapGVK = schema.GroupVersionKind{Version: "v1", Kind: "ConfigMap"}
+	nsGVK        = schema.GroupVersionKind{Version: "v1", Kind: "Namespace"}
+	podGVK       = schema.GroupVersionKind{Version: "v1", Kind: "Pod"}
+)
+
+const (
+	timeout = time.Second * 10
+	tick    = time.Second * 2
+)
+
+// Test_ReconcileSyncSet_wConfigController verifies that SyncSet and Config resources
```

Comment:
Testing this in conjunction with config seems like over-testing.

Testing the set-like behavior of watch intents has already been handled by the cache manager tests.

Functionality we're looking to test:

1) SyncSets can be added/removed from cache manager
2) GVKs described by sync sets show up in the cache.

Testing (2) is optional, if its already covered by the cache manager tests.

In short, any reference to "config" should be removed.